### ISSUE TITLE
createEntrySpan should get SamplingService  only if  it required ( carrier is valid: move #L90 after #L93  )

### ISSUE THREAD

**Issue by qxo** [apache/skywalking#3135]
https://github.com/apache/skywalking/blob/c7916d9f2715ff9d3c415f86418d8f81c97a21e7/apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/ContextManager.java#L90

https://github.com/apache/skywalking/blob/c7916d9f2715ff9d3c415f86418d8f81c97a21e7/apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/ContextManager.java#L93
```
public static AbstractSpan createEntrySpan(String operationName, ContextCarrier carrier) {
        SamplingService samplingService = ServiceManager.INSTANCE.findService(SamplingService.class);
        AbstractSpan span;
        AbstractTracerContext context;
        if (carrier != null && carrier.isValid()) {
            samplingService.forceSampled();
            context = getOrCreate(operationName, true);
            span = context.createEntrySpan(operationName);
            context.extract(carrier);
        } else {
            context = getOrCreate(operationName, false);
            span = context.createEntrySpan(operationName);
        }
        return span;
    }
```

**Comment by wu-sheng** [comment]
Yes, you could move it in the `if`

**Comment by kezhenxu94** [comment]
fixed by #3137
